exercises/react/react.ts

```diff
@@ -0,0 +1,102 @@
+//
+// Delete and replace stub with your own implementation
+//
+// Inspired by "How it Works":
+// https://indepth.dev/posts/1269/finding-fine-grained-reactive-programming#how-it-works
+// https://levelup.gitconnected.com/finding-fine-grained-reactive-programming-89741994ddee?source=friends_link&sk=31c66a70c1dce7dd5f3f4229423ad127#4543
+//
+// and "Computations":
+// https://github.com/ryansolid/solid/blob/master/documentation/reactivity.md#user-content-computations
+//
+
+type EqualFn<T> = (lhs: T, rhs: T) => boolean
+type GetterFn<T> = () => T
+type SetterFn<T> = (v: T) => T
```

Comment:
I'm fine with `Fn`, but I do think we should write out `v` as `value`. Not too sure about `lhs` and `rhs`. _if_ we keep them as `lhs` and `rhs`, perhaps we can add some `tsdoc` to explain them.

```suggestion
type SetterFn<T> = (value: T) => T
```

Reply:
Replaced with:

```typescript
/**
 * Type for the closure's value equality predicate.
 *
 * @typeParam T - Type of the values being compared for
 *              equality.
 *
 * @remarks
 * Conceptually this function should be equivalent
 * to: `lhs === rhs`
 *
 * @param lhs   - left hand side value
 * @param rhs   - right hand side value
 * @returns     - `true` if values are considered
 *                equal; `false` otherwise.
 */
type EqualFn<T> = (lhs: T, rhs: T) => boolean
```